contracts/rust/src/cape/submit_block.rs

```diff
@@ -95,7 +95,18 @@ pub async fn submit_cape_block_with_memos(
     // gas usage of processing a burn note is potentially unbounded. Using
     // tokens whose transfer function far exceeds normal gas consumption is
     // currently not supported.
-    tx.set_gas(gas_limit);
+    //
+    // TODO: mathis: it's a bit wasteful to download the entire block for this
+    // but I don't know of another way to obtain the current block gas limit.
+    let block = contract
+        .client()
+        .get_block(BlockNumber::Latest)
+        .await?
+        .unwrap();
```

Comment:
I will remove this unwrap.

Reply:
https://github.com/EspressoSystems/cape/pull/1230/commits/3480fd8650fc9fd337399184b366787bd398a87b